### PR TITLE
camerad: Out of bounds memory write

### PR DESCRIPTION
**Pull request by geohot**
This may break things, but it's correct.

### CODE REVIEW

**Comment by geohot** [comment]
@adeebshihadeh Are we running things in valgrind?

**Comment by adeebshihadeh** [comment]
> @adeebshihadeh Are we running things in valgrind? (maybe it wouldn't find this anyway since it's a weird allocation)

Only ubloxd right now. It's on the list for the upcoming testing sprint.

File: selfdrive/camerad/cameras/camera_qcom2.cc [hunk 1]
```diff
@@ -227,7 +227,7 @@ void sensors_init(int video0_fd, int sensor_fd, int camera_num) {
   buf_desc[0].size = buf_desc[0].length = sizeof(struct cam_cmd_i2c_info) + sizeof(struct cam_cmd_probe);
   buf_desc[0].type = CAM_CMD_BUF_LEGACY;
   struct cam_cmd_i2c_info *i2c_info = (struct cam_cmd_i2c_info *)alloc_w_mmu_hdl(video0_fd, buf_desc[0].size, (uint32_t*)&buf_desc[0].mem_handle);
-  struct cam_cmd_probe *probe = (struct cam_cmd_probe *)((uint8_t *)i2c_info) + sizeof(struct cam_cmd_i2c_info);
+  struct cam_cmd_probe *probe = (struct cam_cmd_probe *)((uint8_t *)i2c_info + sizeof(struct cam_cmd_i2c_info));
```
Comment (deanlee) [inline selfdrive/camerad/cameras/camera_qcom2.cc:230]: Would this be clearer and less error-prone:
`auto probe = (struct cam_cmd_probe *)(i2c_info + 1);`

**Comment by geohot** [comment]
I'm bringing up the IMX390 and it was causing issues with that sensor. It seems okay with the AR0231, but the version check doesn't work. (it's reading 0)

File: selfdrive/camerad/cameras/camera_qcom2.cc (hunk 1 above)
Comment (geohot) [reply selfdrive/camerad/cameras/camera_qcom2.cc:230]: Good point

**Comment by geohot** [comment]
Should be good we merge if we trust that CI checks cameras

**Pull request merged**